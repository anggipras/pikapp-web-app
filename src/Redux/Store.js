import { createStore } from 'redux'
// import Thunk from 'redux-thunk'
import Reducers from './Reducers'

const store = createStore(Reducers)

export default store