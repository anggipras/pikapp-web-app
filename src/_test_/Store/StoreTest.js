import Axios from "axios";
import { address, clientId } from "../../Asset/Constant/APIConstant";
import { v4 as uuidV4 } from "uuid";

export const fetchData = async () => {
    let latitude = -6.28862
    let longitude = 106.71789
    let addressRoute = address + "home/v2/merchant/" + longitude + "/" + latitude + "/ALL/"

    let uuid = uuidV4();
    uuid = uuid.replace(/-/g, "");
    const date = new Date().toISOString();

    const allMerchantAPI = await Axios(addressRoute, {
        headers: {
            "Content-Type": "application/json",
            "x-request-id": uuid,
            "x-request-timestamp": date,
            "x-client-id": clientId,
            "token": "PUBLIC",
            "category": "1",
        },
        method: "GET",
        params: {
            page: 0,
            size: 6
        }
    })

    let allData = allMerchantAPI.data.results
    let realData = []
    realData.push({
        address: allData.merchant_address,
        rating: allData.merchant_rating,
        logo: allData.merchant_logo,
        distance: allData.merchant_distance,
        storeId: allData.mid,
        storeName: allData.merchant_name,
        storeDesc: "",
        storeImage: allData.merchant_pict,
    })

    return realData
}

export const loadMoreMerch = () => {
    let idCol = 0
    let page = 0 //will load more merchant until all pages load
    let totalpage = 7 //there are 6 pages for instance

    do {
        if (idCol <= page) {
            idCol += 1
            page += 1
            console.log('loadagain');
        } else {
            console.log('disable scroll');
        }
    } while (page < totalpage - 1);

    return page
}