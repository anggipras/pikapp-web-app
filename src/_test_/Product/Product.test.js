import { shallow } from "enzyme";
import { Provider } from 'react-redux'
import Store from '../../Redux/Store'
import ProductView from "../../View/Product/ProductView";
import { BrowserRouter as Router } from "react-router-dom";
import { fetchMerchantDetail, handleAddCart } from './ProductTest'
import renderer from 'react-test-renderer'

it("renders without crashing ProductView", () => {
  shallow(
    <Provider store={Store}>
      <ProductView />
    </Provider>
  );
});

it('matches the snapshot ProductView', () => {
  const productTree = renderer.create(
    <Router>
      <Provider store={Store}>
        <ProductView />
      </Provider>
    </Router>
  ).toJSON()

  expect(productTree).toMatchSnapshot()
})

describe('fetchMerchantDetail', () => {
  it('render merchant detail data correctly', () => {
    let merchantDetailImit = {
      "pickups": null,
      "products": [
        {
          "product_name": "Mie Ayam Yamin Jumbo Pangsit",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A30%3A01DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=9ac298c9797cb4827aa84613dde88cae39bd6388c4b7270941e1076c24dd6960",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A30%3A01DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=0df9b7de9fd35c36c0cf47042122c819a5960d18b8fbddd69782b7adc9f8ef5c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A30%3A01DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=0df9b7de9fd35c36c0cf47042122c819a5960d18b8fbddd69782b7adc9f8ef5c",
          "product_price": 30000,
          "product_id": "P00000026",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + pangsit kuah",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Ayam Yamin Jumbo Bakso",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000027",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Ayam Yamin Jumbo Kecap",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000028",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Ayam Yamin Jumbo Bakso Bakso",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=f92ce5615146f1e8dc0129531062571d23b784201a4915abf9ddbed5a90ba459",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000029",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Ayam",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000030",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Coba Coba",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000031",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Coba 01",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000032",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Coba 02",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000033",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Coba 03",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000034",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie Coba 05",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=f92ce5615146f1e8dc0129531062571d23b784201a4915abf9ddbed5a90ba459",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000037",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        },
        {
          "product_name": "Mie 007 ",
          "product_picture1": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture2": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_picture3": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A31%3A13DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T093852Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=7e95c3de6b1a361ec1be5c0081bdb79fe4b1aa6606522a4cde52be68b45c585c",
          "product_price": 30000,
          "product_id": "P00000038",
          "product_status": "true",
          "product_category": 4,
          "product_desc": "Mie ukuran besar ditambah bumbu kecap dengan potongan daging ayam dan kriuk ayam yang renyah + bakso",
          "rating": "0.0"
        }
      ],
      "categories": [
        {
          "category_id": "4",
          "category_name": "Mie",
          "order": 0
        }
      ],
      "mid": "M00000008",
      "merchant_pict": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T103508Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=9c99aec227de7a54c7239d394a5d68ac5c1b943ca784787d6267cb8fd060e346",
      "merchant_logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210521T103508Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210521%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=a47cc09b0631eac741857da77bbb44d92dbf62edc6548a7b3bdaefaada49d000",
      "merchant_name": "Pikapp Store",
      "merchant_level": null,
      "merchant_rating": null,
      "merchant_address": "Jl. Regensi Melati Mas F10/2",
      "merchant_distance": "7.61 km"
    }

    fetchMerchantDetail().then(merchDet => {
      expect(merchDet).toEqual(merchantDetailImit)
    })
  })
})

it('render addtocart merchant detail', () => {
  var cart = [
    {
      mid: "",
      storeName: "",
      storeDesc: "",
      storeDistance: "",
      food: [
        {
          productId: "",
          foodName: "",
          foodPrice: 0,
          foodAmount: 0,
          foodImage: "",
          foodNote: "",
        },
      ],
    },
    {
      mid: "M00000008",
      storeName: "Pikapp Store",
      storeDesc: "Desc",
      storeDistance: "7.61 km",
      food: [
        {
          productId: "P00000026",
          foodName: "Mie Ayam Yamin Jumbo Pangsit",
          foodPrice: 30000,
          foodAmount: 1,
          foodImage: "https://production-merchant.s3.ap-southeast-1.amazonaws.com/09-04-2021_10%3A30%3A01DSCF6013-2%202.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210524T091543Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210524%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=636bc139a2f7f583738d963d4f4d9eac4a876cf0ba4c216234eb633e16f7e9ee",
          foodNote: "",
        },
      ],
    }
  ];

  let incomingData = handleAddCart()
  expect(cart).toEqual(incomingData)
})