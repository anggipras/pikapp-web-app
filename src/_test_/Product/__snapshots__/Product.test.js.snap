// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot ProductView 1`] = `
Array [
  <div
    className="storeBanner"
  >
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={
          Object {
            "height": "100%",
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        ‌
      </span>
    </span>
    <div />
  </div>,
  <div
    className="merchant-section"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <div
      className="inside-merchantSection"
    >
      <div
        className="merchant-info"
      >
        <div
          className="top-merchantInfo"
        >
          <div
            className="inside-topMerchantInfo"
          >
            <div
              className="merchant-title"
            >
              <div
                className="merchant-logo"
              >
                <span>
                  <span
                    className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                    style={
                      Object {
                        "height": "100%",
                        "paddingTop": 10,
                        "width": "100%",
                      }
                    }
                  >
                    ‌
                  </span>
                </span>
              </div>
              <div
                className="merchant-name"
              >
                <div
                  className="merchant-mainName"
                >
                  <span>
                    <span
                      className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style={
                        Object {
                          "paddingTop": 30,
                          "width": 200,
                        }
                      }
                    >
                      ‌
                    </span>
                  </span>
                </div>
                <div
                  className="merchant-categName"
                >
                  <div
                    className="merchant-allcateg"
                  >
                    Merchant Category
                  </div>
                  <div
                    className="merchant-starInfo"
                  />
                </div>
              </div>
            </div>
            <div
              className="merchant-call-sec"
              onClick={[Function]}
            >
              <div
                className="merchant-call"
              >
                <span
                  className="merchantCall-icon"
                >
                  <img
                    alt=""
                    className="merchantCall-img"
                    src="phone.png"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="bottom-merchantInfo"
        >
          <div
            className="inside-bottomMerchantInfo"
          >
            <div
              className="merchantdetail-section"
            >
              <div
                className="icon-based"
              >
                <img
                  alt=""
                  className="locationicon"
                  src="location.png"
                />
              </div>
              <div
                className="detail-info"
              >
                <div
                  className="top-detail-info"
                >
                  Store Address
                </div>
                <div
                  className="bottom-detail-info"
                >
                  <span>
                    <span
                      className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style={
                        Object {
                          "paddingTop": 30,
                          "width": 100,
                        }
                      }
                    >
                      ‌
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="merchant-category"
      >
        <div
          className="select-category"
        >
          <div
            className="listCategory"
          >
            <h2
              className="categoryName"
            >
              All Categories
            </h2>
            <div
              className="arrow-based"
              onClick={[Function]}
            >
              <img
                alt=""
                className="arrowicon"
                src="arrowselect.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="product-layout"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <div
      className="mainproduct-sec"
    >
      <div
        className="product-section"
      >
        <h2
          className="product-categ"
          id=""
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={
                Object {
                  "height": 50,
                  "width": 200,
                }
              }
            >
              ‌
            </span>
          </span>
        </h2>
        <div
          className="list-product"
        />
        <div
          id={0}
        />
      </div>
      <div
        className="pikapp-info"
      >
        <h3
          className="pikappInfo"
        >
          Digital Menu By
        </h3>
        <img
          alt=""
          className="Logopikapp"
          src="logo4x.png"
        />
      </div>
    </div>
  </div>,
]
`;
